### ISSUE TITLE
Compensate for backlash

### ISSUE THREAD

**Issue by bgottula** [bgottula/track#8]
Estimation of backlash is covered in #7. For this issue assume that the amount of backlash in azimuth and elevation are known quantities. The problem to be solved is how to compensate for the backlash when tracking.

For optical tracking it may be possible to get away with no backlash compensation although the target may stray from the center of the field of view temporarily if the slew direction changes in the middle of a pass. If the backlash is bad enough the target may even wander completely beyond the field of view before the control loop has time to react.

For ephemeris based tracking backlash definitely needs to be corrected. One method would be to add a correction factor to the azimuth/altitude coordinates reported by the mount depending on the slew direction before using them to compute the pointing error. This would need to account for which approach direction was used during alignment. This would create an abrupt change in the pointing error signal whenever the slew direction changes. The control loop would eventually bring the error back to zero but the response time will be slow and will probably allow the target to wander outside the field of view until it catches up. In addition to correcting the reported position, it may be possible to temporarily "slew" at a high rate until the backlash has been compensated, after which the slew rate would be restored to its previous rate.

Ideally any corrections for backlash would be applied in the TelescopeMount class.

**Comment by bgottula** [comment]
Scenarios for backlash compensation within the mount class:

- Slewing in the same direction used during alignment: The position reported by the mount should be the true position. No correction needs to be applied.
- Slewing in the opposite direction:
   - Immediately after changing slew direction, but before the optical tube has started moving: The mount will report that the position has moved in the direction of the slew, but the actual position has remained unchanged. Ideally the mount class should report no change in position, however it may not be essential to handle this special case. Handling this case would require knowledge of the mount's reported position at the moment the slew direction changed and this information may not be recorded.
   - Optical tube has started moving again: The actual position will lag the mount's reported position by the backlash amount. Mount class should report the mount's reported position compensated by the backlash estimate.

**Comment by bgottula** [comment]
Tested backlash compensation with the script test_backlash_compensation.py. I ran three cases, one with no compensation applied, one with 1000 arcseconds deadband in both axes with a positive approach direction, and another with 1000 arcseconds backlash with negative approach direction. The results are attached and confirm that compensation is working as expected.
[backlash_test_data.zip](https://github.com/bgottula/track/files/1220287/backlash_test_data.zip)
